connections/ethereum/gsn/gsn.go

```diff
@@ -36,49 +34,56 @@ type gsnResponse struct {
 	GasPriceRange interface{} `json:"gasPriceRange"`
 }
 
-// callGSN will call the Gas Station Network and request the gas prices on the Ethereum network
-func CallGSN(apiKey, speed string) (error, *big.Int) {
-	time.Sleep(1 * time.Second)
-
+// FetchGasPrice will query GSN for the current gas prices and return the price for the specified speed.
+func FetchGasPrice(apiKey, speed string) (*big.Int, error) {
 	var gsnURL = "https://ethgasstation.info/api/ethgasAPI.json?api-key=" + apiKey
 
-	var gsnRes gsnResponse
-
-	retries := 10
-	for i := 0; i < retries; i++ {
-		client := &http.Client{Timeout: time.Second * 10}
-		res, err := client.Get(filepath.Clean(gsnURL))
-		if err != nil {
-			return err, nil
-		}
-		defer res.Body.Close()
-
-		body, err := ioutil.ReadAll(res.Body)
-		if err != nil {
-			return err, nil
-		}
-		err = json.Unmarshal(body, &gsnRes)
+	for i := 0; i < Retries; i++ {
+		res, err := queryAPI(gsnURL)
 		if err != nil {
-			return err, nil
+			return nil, err
 		}
 
-		if body != nil && res.StatusCode == http.StatusOK {
-			break
+		if res != nil {
+			return parsePrice(res, speed), nil
 		}
 
 		time.Sleep(1 * time.Second)
 	}
 
-	return nil, getPrice(&gsnRes, speed)
+	return nil, errors.New("failed to fetch GSN gas price - retries exceeded")
```

Comment:
Maybe its better to use some default value here? Bcs it looks like bridge became totally dependent on ethgasstation api

Reply:
If no price is returned it will fallback to eth_estimateGas https://github.com/ChainSafe/ChainBridge/pull/628/files#diff-349cc0f2c50c806250ed335c780d7a6dd9b7ae1cd01fed7e8cfacfa16bba81f6R138-R155